### PR TITLE
Implement hashCode, equals and toString in Maybe

### PR DESCRIPTION
**Pull request by Duhemm**
This brings Maybe's behavior closer to scala's Option.

### CODE REVIEW

**Comment by dwijnand** [comment]
LGTM

File: internal/util-interface/src/main/java/xsbti/Maybe.java [hunk 1]
```diff
@@ -14,17 +14,36 @@ protected Maybe() {}
 		return new Maybe<s>() {
 			public boolean isDefined() { return true; }
 			public s get() { return v; }
+			public int hashCode() { return 17 + (v == null ? 0 : v.hashCode()); }
+			public String toString() { return "Maybe(" + v + ")"; }
```
Comment (eed3si9n) [inline internal/util-interface/src/main/java/xsbti/Maybe.java]: Shouldn't this be `Maybe.just(" + v + ")` so `toString` output corresponds with the Scala code to create the value?

File: internal/util-interface/src/main/java/xsbti/Maybe.java [hunk 2]
```diff
@@ -14,17 +14,36 @@ protected Maybe() {}
...
+			}
 		};
 	}
 	public static <s> Maybe<s> nothing()
 	{
 		return new Maybe<s>() {
 			public boolean isDefined() { return false; }
 			public s get() { throw new UnsupportedOperationException("nothing.get"); }
+			public int hashCode() { return 1; }
+			public String toString() { return "Nothing"; }
```
Comment (eed3si9n) [inline internal/util-interface/src/main/java/xsbti/Maybe.java]: Similarly, `Maybe.nothing()`.

**Comment by Duhemm** [comment]
That's a good idea. I've pushed a new commit.

**Pull request merged**